client/post-editor/editor-page-templates/index.jsx

```diff
@@ -1,92 +1,106 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
-import find from 'lodash/find';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { find, size, map } from 'lodash';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import PostActions from 'lib/posts/actions';
+import QueryPageTemplates from 'components/data/query-page-templates';
 import AccordionSection from 'components/accordion/section';
 import SelectDropdown from 'components/select-dropdown';
 import DropdownItem from 'components/select-dropdown/item';
+import EditorDrawerLabel from 'post-editor/editor-drawer/label';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
+import { getPageTemplates } from 'state/page-templates/selectors';
+import { editPost } from 'state/posts/actions';
 
-export default React.createClass( {
-	displayName: 'EditorPageTemplates',
+class EditorPageTemplates extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		postId: PropTypes.number,
+		postType: PropTypes.string,
+		template: PropTypes.string,
+		templates: PropTypes.array,
+		editPost: PropTypes.func
+	};
 
-	propTypes: {
-		post: PropTypes.object,
-		pageTemplates: PropTypes.array.isRequired
-	},
-
-	getDefaultProps() {
-		return {
-			pageTemplates: []
-		};
-	},
+	constructor() {
+		super( ...arguments );
 
-	render() {
-		if ( ! this.props.pageTemplates.length || this.props.pageTemplates.length === 1 ) {
-			return null;
-		}
-		return (
-			<div>
-				<label className="editor-drawer__label">
-					<span className="editor-drawer__label-text">{ this.translate( 'Page Template' ) }</span>
-				</label>
-				{ this._renderTemplateOptions() }
-			</div>
-		);
-	},
-
-	_renderTemplateOptions() {
-		return (
-			<AccordionSection>
-				<SelectDropdown selectedText={ this._getSelectedTemplateText() }>
-					{
-						this.props.pageTemplates.map( ( template, i ) => {
-							return (
-								<DropdownItem
-									key={ 'page-templates-' + i }
-									selected={ this._isTemplateSelected( template ) }
-									onClick={ this._selectTemplate.bind( this, template ) }
-								>
-									{ template.label }
-								</DropdownItem>
-							);
-						} )
-					}
-				</SelectDropdown>
-			</AccordionSection>
-		);
-	},
-
-	_isTemplateSelected( template ) {
-		if ( ! this.props.post ) {
-			return false;
-		}
-		return template.file === this.props.post.page_template;
-	},
+		this.selectTemplate = this.selectTemplate.bind( this );
+	}
 
-	_selectTemplate( template ) {
-		// TODO: REDUX - remove flux actions when whole post-editor is reduxified
-		PostActions.edit( { page_template: template.file } );
-	},
+	selectTemplate( file ) {
+		const { siteId, postId } = this.props;
+		this.props.editPost( { page_template: file }, siteId, postId );
```

Comment:
>I definitely think we should change the order of the arguments to match the typical selector flow

Yeah, I'm inclined to agree. Will take a look at refactoring this in a later pull request.